modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala

```diff
@@ -1027,8 +1037,12 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
             hints(member) ++ default ++ noDefault
           )
         }
-        .collect { case (name, Some(tpe), required, hints) =>
-          Field(name, tpe, required, hints)
+        .collect {
+          case (name, Some(tpe: Type.ExternalType), required, hints) =>
+            val newHints = hints.filterNot(_ == tpe.refinementHint)
+            Field(name, tpe, required, newHints)
```

Comment:
question: is it enough to do it on fields? I may not understand `Field` correctly, but I'd guess that we should do this everywhere we do `hints(shape)`

Reply:
Yeah that's a good call, I will check it out